### PR TITLE
:sparkles:(lld): add actions in ordis drawer

### PR DESCRIPTION
**Pull request by LucasWerey**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - NewArch nft drawer (ordis too)

### 📝 Description

Refactor hooks that implements the actions inside nft/ordis drawer so it works using our nft metadata and simplehash API
Inscriptions can be seen on magicEden / ordinals.com or set as custom lock screen on stax/flex

https://github.com/user-attachments/assets/0d149b8d-3f7a-4e6c-bb16-ab8a8644acd3

### ❓ Context

- **JIRA or GitHub link**: [LIVE-14261](https://ledgerhq.atlassian.net/browse/LIVE-14261)(

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-14261]: https://ledgerhq.atlassian.net/browse/LIVE-14261?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        .map(link => ({
+          id: link.marketplace_id,
+          label: t("NFT.viewer.actions.open", { viewer: link.marketplace_name }),
+          Icon: IconMagicEden,
+          type: ItemType.EXTERNAL,
+          callback: () => openURL(link.nft_url),
+        })),
+
+      ...(marketplacePages.length > 0 ? [{ id: "sep2", type: "separator", label: "" }] : []),
+      explorerLink && {
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx:36]: if `explorerLink` is falsy, you will get an unwanted `false` value in your array

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/Inscriptions/DetailsDrawer/index.tsx [hunk 2]
```diff
@@ -5,19 +5,23 @@ import { CollectibleTypeEnum } from "LLD/features/Collectibles/types/enum/Collec
 import useInscriptionDetailDrawer from "./useInscriptionDetailDrawer";
 import Actions from "./Actions";
 import SubTitle from "./SubTitle";
+import { BitcoinAccount } from "@ledgerhq/coin-bitcoin/lib/types";
 
 type ViewProps = ReturnType<typeof useInscriptionDetailDrawer> & {
+  inscription: SimpleHashNft;
```
[the remark this answers is not in the document]
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/Inscriptions/DetailsDrawer/index.tsx:11]: No real reason just find it more explicit to do this way. So we know that we don't change account in the Model

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/screens/Account/index.tsx [hunk 3]
```diff
@@ -7,7 +7,9 @@ import { BitcoinAccount } from "@ledgerhq/coin-bitcoin/lib/types";
 import { useBitcoinAccountModel } from "./useBitcoinAccountModel";
 import InscriptionDetailsDrawer from "../../components/Inscriptions/DetailsDrawer";
 
-type ViewProps = ReturnType<typeof useBitcoinAccountModel>;
+type ViewProps = ReturnType<typeof useBitcoinAccountModel> & {
```
[the remark this answers is not in the document]
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/screens/Account/index.tsx:10]: No real reason just find it more explicit to do this way. So we know that we don't change account in the Model

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        id: ExternalIdEnum.EXPLORER,
+        label: t("NFT.viewer.actions.open", { viewer: "Explorer" }),
+        Icon: IconsLegacy.GlobeMedium,
+        type: ItemType.EXTERNAL,
+        callback: () => openURL(explorerLink),
+      },
+    ]);
+  } else {
+    const linksRecord = links as Record<string, string>;
+    return safeList([
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx:46]: same here , avoid having a `[false, false, false]` or mixed with false array type

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import { ExternalIdEnum, ItemType } from "LLD/features/Collectibles/types/enum/Links";
+import IconOpensea from "~/renderer/icons/Opensea";
+import IconRarible from "~/renderer/icons/Rarible";
+import IconMagicEden from "~/renderer/icons/MagicEden";
+import { IconsLegacy } from "@ledgerhq/react-ui";
+import { SimpleHashNft } from "@ledgerhq/live-nft/api/types";
+import { NFTMetadata, ProtoNFT } from "@ledgerhq/types-live";
+import { isNFTMetadata } from "../utils/typeGuardsChecker";
+import { createOrdinalExplorerUrl } from "../utils/createOrdinalExplorerUrl";
+
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx:15]: (suggestion) can you summarise in few words what this hook stands for please?

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        type: ItemType.EXTERNAL,
+        callback: () => openURL(explorerLink),
+      },
+    ]);
+  } else {
+    const linksRecord = links as Record<string, string>;
+    return safeList([
+      linksRecord.opensea && {
+        id: ExternalIdEnum.OPENSEA,
+        label: t("NFT.viewer.actions.open", { viewer: "Opensea.io" }),
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx:49]: `viewer:` different values can be extract in a shared enum

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useHideCollection.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import { SimpleHashNft } from "@ledgerhq/live-nft/api/types";
+import { IconsLegacy } from "@ledgerhq/react-ui";
+import { ProtoNFT, NFTMetadata, Account } from "@ledgerhq/types-live";
+import { TFunction } from "i18next";
+import { useMemo } from "react";
+import { useDispatch } from "react-redux";
+import { openModal } from "~/renderer/actions/modals";
+import { isNFTMetadata } from "../utils/typeGuardsChecker";
+
+export const useHideCollection = (
```
[the remark this answers is not in the document]
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useHideCollection.ts:10]: The hook is here to open the hide collection modal so I am not sure that hiddenCollection would be clearer. At least I can rename it to useOpenHideCollectionModal

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx (hunk 6 above)
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx:49]: new Enum has been added ✅

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useHideCollection.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (themooneer) [reply apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useHideCollection.ts]: okey ! sounds good for me

**Review by themooneer — APPROVED** [review]

**Review by thesan — APPROVED** [review]

**Pull request merged**